shard-operator/src/main/java/com/redhat/service/bridge/shard/operator/controllers/BridgeIngressController.java

```diff
@@ -35,10 +41,36 @@
     @Inject
     ManagerSyncService managerSyncService;
 
+    @Inject
+    BridgeIngressService bridgeIngressService;
+
+    @Override
+    public void init(EventSourceManager eventSourceManager) {
+        DeploymentEventSource deploymentEventSource = DeploymentEventSource.createAndRegisterWatch(kubernetesClient, LabelsBuilder.BRIDGE_INGRESS_APPLICATION_TYPE);
+        eventSourceManager.registerEventSource("bridge-ingress-deployment-event-source", deploymentEventSource);
+    }
+
     @Override
     public UpdateControl<BridgeIngress> createOrUpdateResource(BridgeIngress bridgeIngress, Context<BridgeIngress> context) {
         // simplistic reconciliation to check with IT
         LOGGER.info("Create or update BridgeIngress: '{}' in namespace '{}'", bridgeIngress.getMetadata().getName(), bridgeIngress.getMetadata().getNamespace());
+
+        Deployment deployment = bridgeIngressService.fetchOrCreateBridgeIngressDeployment(bridgeIngress);
+
+        if (!Readiness.isDeploymentReady(deployment)) {
+            LOGGER.info("Ingress deployment BridgeIngress: '{}' in namespace '{}' is NOT ready", bridgeIngress.getMetadata().getName(),
+                    bridgeIngress.getMetadata().getNamespace());
+            return UpdateControl.noUpdate();
```

Comment:
Are we considering updating the `BridgeIngress` status here? `READY: false`. On Knative we copy the `deployment` reference status to set this info.

Reply:
Hey @ricardozanini , yes it's a good idea. 

I had a look at the cos-fleetshard and they are using a `PhaseType` property in the Spec to keep track of the status. I've adopted the same approach